components/api/src/main/java/com/hotels/styx/api/configuration/ServiceFactory.java

```diff
@@ -32,4 +32,16 @@
      * @return product instance
      */
     E create(Environment environment, Configuration serviceConfiguration);
+
+    /**
+     * Create an instance of the product.
+     *
+     * @param environment                 environment
+     * @param serviceConfiguration configuration specific to the factory product
+     * @param objects objects that service might depend on
+     * @return product instance
+     */
+    default E create(Environment environment, Configuration serviceConfiguration, Object... objects) {
```

Comment:
Let's get rid of the `Objects...` array. However I suppose this is also used for the styx "server" class loading? In that case just duplicate this interface for now. Create a separate variant for "StyxClientService" etc which always takes an OriginInventory as a third parameter. Going forward we will remove the duplicated interface in the next increment. But at the moment we'll need running code.